addons/event_sale/models/event.py

```diff
@@ -59,7 +59,7 @@ def _is_event_registrable(self):
         self.ensure_one()
         if not self.event_ticket_ids:
             return True
-        return all(self.event_ticket_ids.with_context(active_test=False).mapped(lambda t: t.product_id.active))
+        return bool(self.event_ticket_ids.with_context(active_test=False).filtered(lambda t: t.product_id.active and not t.is_expired and t.seats_available))
```

Comment:
You should consider the case when `seats_max` is 0, wich for Odoo mean unlimited seats

Reply:
Maybe something like:
```suggestion
        return bool(self.event_ticket_ids.with_context(active_test=False).filtered(lambda t: t.product_id.active and not t.is_expired and t.seats_available if t.seats_max else True))
```